### PR TITLE
More preparations for Scala 3

### PR DESCRIPTION
**Pull request by ybasket**
Rewrites lambda syntax for FunctionK so that these snippets cross compile on Scala 2 & 3. Fixes a visibility issue with span implementations and adds parentheses around lambda parameters as Scala 3 requires them.

### CODE REVIEW

File: modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Lift.scala [hunk 1]
```diff
@@ -7,7 +7,9 @@ trait Lift[Low[_], F[_]] extends ContextRoot { self =>
   def F: Monad[F]
 
   def lift[A](la: Low[A]): F[A]
-  def liftK: Low ~> F = λ[Low ~> F](lift(_))
+  def liftK: Low ~> F = new ~>[Low, F] {
```
Comment (catostrophe) [inline modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Lift.scala:10]:
```suggestion
  def liftK: Low ~> F = new (Low ~> F) {
```

File: modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Local.scala [hunk 2]
```diff
@@ -5,10 +5,14 @@ import io.janstenpickle.trace4cats.base.optics.Lens
 
 trait Local[F[_], R] extends Ask[F, R] { self =>
   def local[A](fa: F[A])(f: R => R): F[A]
-  def localK(f: R => R): F ~> F = λ[F ~> F](local(_)(f))
+  def localK(f: R => R): F ~> F = new ~>[F, F] {
```
Comment (catostrophe) [inline modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Local.scala:8]:
```suggestion
  def localK(f: R => R): F ~> F = new (F ~> F) {
```

File: modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Local.scala [hunk 3]
```diff
@@ -5,10 +5,14 @@ import io.janstenpickle.trace4cats.base.optics.Lens
...
+    override def apply[A](fa: F[A]): F[A] = local(fa)(f)
+  }
 
   def scope[A](fa: F[A])(r: R): F[A] = local(fa)(_ => r)
-  def scopeK(r: R): F ~> F = λ[F ~> F](scope(_)(r))
+  def scopeK(r: R): F ~> F = new ~>[F, F] {
```
Comment (catostrophe) [inline modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Local.scala:13]:
```suggestion
  def scopeK(r: R): F ~> F = new (F ~> F) {
```

File: modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Provide.scala [hunk 4]
```diff
@@ -5,7 +5,9 @@ import cats.{~>, Monad}
 trait Provide[Low[_], F[_], R] extends Local[F, R] with Unlift[Low, F] { self =>
   def provide[A](fa: F[A])(r: R): Low[A]
 
-  def provideK(r: R): F ~> Low = λ[F ~> Low](provide(_)(r))
+  def provideK(r: R): F ~> Low = new ~>[F, Low] {
```
Comment (catostrophe) [inline modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Provide.scala:8]:
```suggestion
  def provideK(r: R): F ~> Low = new (F ~> Low) {
```

File: modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Lift.scala [hunk 5]
```diff
@@ -7,7 +7,9 @@ trait Lift[Low[_], F[_]] extends ContextRoot { self =>
...
+    override def apply[A](fa: Low[A]): F[A] = lift(fa)
```
Comment (catostrophe) [inline modules/base/src/main/scala/io/janstenpickle/trace4cats/base/context/Lift.scala:11]:
```suggestion
    override def apply[A](la: Low[A]): F[A] = lift(la)
```

**Review by catostrophe — COMMENTED** [review]
@ybasket Nice!

Btw, how do you check the compatibility with Scala 3? Via `-Xsource:3` or directly with `scalac` 3.0.0?

**Review by catostrophe — APPROVED** [review]

**Comment by ybasket** [comment]
> Btw, how do you check the compatibility with Scala 3? Via `-Xsource:3` or directly with `scalac` 3.0.0?

I plugged in 3.0.1-RC2 to `ci-release.sbt` and disable some dependencies (kittens, scalacheck-shapeless) to compile as many modules as possible. That worked actually quite well, with providing hand-written `Eq` instances (by converting to tuples, seems to be the most concise), all modules' main code compiled. Kittens seems to take a while, so maybe that's a way to unblock Scala 3 development here (kittens would be easy to re-introduce). And for scalacheck-shapeless, I saw Michael Pilquist asking on Discord (#shapeless) with some scalacheck derivation code, so maybe there's a solution emerging as well, at least for the modules of this repo.

**Pull request merged**